File: tripnest-backend/src/main/java/com/tripnest/backend/service/TripService.java
package com.tripnest.backend.service;

import com.tripnest.backend.entity.Trip;
import com.tripnest.backend.repository.TripRepository;
import org.springframework.stereotype.Service;

import java.util.List;

import com.tripnest.backend.entity.User;
import com.tripnest.backend.repository.UserRepository;

@Service
public class TripService {

    private final TripRepository tripRepository;
    private final UserRepository userRepository;

    public TripService(TripRepository tripRepository,
                    UserRepository userRepository) {

        this.tripRepository = tripRepository;
        this.userRepository = userRepository;
    }

    // Create a new trip
    public Trip createTrip(Trip trip, String email) {

        User user = userRepository.findByEmail(email)
                .orElseThrow(() -> new RuntimeException("User not found"));

        trip.setUser(user);

        return tripRepository.save(trip);
    }

    // Get all trips
    public List<Trip> getAllTrips(String email) {

        User user = userRepository.findByEmail(email)
                .orElseThrow(() -> new RuntimeException("User not found"));

        return tripRepository.findByUser(user);

    }

    // Get trips by user
    public List<Trip> getTripsByUserId(Long userId) {
        return tripRepository.findByUserId(userId);
    }

    // Get Trip by ID
    public Trip getTripById(Long id) {
        return tripRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Trip not found"));
    }

    // Update Trip
    public Trip updateTrip(Long id, Trip updatedTrip) {

        Trip trip = getTripById(id);

        trip.setTitle(updatedTrip.getTitle());
        trip.setDestination(updatedTrip.getDestination());
        trip.setStartDate(updatedTrip.getStartDate());
        trip.setEndDate(updatedTrip.getEndDate());
        trip.setBudget(updatedTrip.getBudget());
        trip.setStatus(updatedTrip.getStatus());
        trip.setDescription(updatedTrip.getDescription());

        return tripRepository.save(trip);
    }

    // Delete Trip
    public void deleteTrip(Long id) {

        Trip trip = getTripById(id);

        tripRepository.delete(trip);
    }

}

File: tripnest-frontend/src/services/tripService.js
import api from "./api";

// Get all trips
export const getAllTrips = async () => {
    const response = await api.get("/trips");
    return response.data;
};

// Create trip
export const createTrip = async (tripData) => {
    const response = await api.post("/trips", tripData);
    return response.data;
};

// Get trip by ID
export const getTripById = async (id) => {
    const response = await api.get(`/trips/${id}`);
    return response.data;
};

// Update trip
export const updateTrip = async (id, tripData) => {
    const response = await api.put(`/trips/${id}`, tripData);
    return response.data;
};

// Delete trip
export const deleteTrip = async (id) => {
    await api.delete(`/trips/${id}`);
};